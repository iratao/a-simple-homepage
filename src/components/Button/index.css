.button {
  padding: 0 12px;
  color: rgb(85, 85, 85);
  font-weight: 400;
  background: white;
  white-space: nowrap;
  text-align: center;
  border: 1px solid grey;
  cursor: pointer;
  user-select: none;
  font-family: inherit;
}

.button[disabled], .button:hover[disabled] {
  color: #00000040;
  border-color: #d9d9d9;
  background: #f5f5f5;
  text-shadow: none;
  box-shadow: none;
  cursor: not-allowed;
}

.button:hover {
  border-color: black;
  color: black;
}

.large {
  height: 44px;
  font-weight: 600;
}

.medium {
  height: 36px;
  font-weight: 500;
}

.small {
  height: 32px;
  font-weight: 400;
}

.loading {
  text-align: center;
  vertical-align: -0.4em;
  width: 1.5em;
}