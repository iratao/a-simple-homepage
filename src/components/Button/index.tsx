import React from 'react'
import CommonProps from '../../types/CommonProps'
import LoadingIcon from '../LoadingIcon'
import styles from './index.less'

interface Props extends CommonProps {
  size?: 'small' | 'medium' | 'large' | 'auto';
  onClick?: React.MouseEventHandler<HTMLElement>;
  className?: string;
  loading?: boolean;
}

const Button = ({ children, size = 'medium', onClick, className = '', loading }: Props) => {
  const disabled = loading
  return (
    <button className={`${className} ${styles.button} ${styles[size]}`} disabled={disabled} onClick={onClick}>
      {loading && <LoadingIcon className={`${styles.loading}`}/>} {children}
    </button>
  )
}

export default Button