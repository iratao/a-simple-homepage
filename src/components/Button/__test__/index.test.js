import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button>Click</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('can set custom className', () => {
  const tree = renderer
    .create(<Button className='myCustomClassName'>Click</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('show loading icon when in loading mode', () => {
  const tree = renderer
    .create(<Button loading={true}>Click</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})