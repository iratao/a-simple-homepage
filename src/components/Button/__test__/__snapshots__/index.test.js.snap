// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can set custom className 1`] = `
<button
  className="myCustomClassName button medium"
>
   
  Click
</button>
`;

exports[`renders correctly 1`] = `
<button
  className=" button medium"
>
   
  Click
</button>
`;

exports[`show loading icon when in loading mode 1`] = `
<button
  className=" button medium"
  disabled={true}
>
  <img
    className="img loading "
    src="test-file-stub"
  />
   
  Click
</button>
`;
